File: C#/0929. Unique Email Addresses.cs
public class Solution {
    public int NumUniqueEmails(string[] emails) {
        HashSet<string> emailSet=new HashSet<string>();
        foreach(string email in emails){
            string curEmail=FixEmail(email);
            emailSet.Add(curEmail);
        }
        return emailSet.Count;
    }
    public string FixEmail(string email){
        string[] EmailSplit=email.Split('@');
        string rep="";
        foreach(char num in EmailSplit[0]){
            if (num!='+' && num!='.'){
                rep+=num;
            }
            else if(num=='+'){
                break;
            }
            else if(num=='.'){
                continue;
            }
        }
        return rep+'@'+EmailSplit[1];
    }
}

File: archives/leetcode2/0929. Unique Email Addresses.py
class Solution:
    def numUniqueEmails(self, emails: 'List[str]') -> 'int':
        k=0
        while k<len(emails):
            emails[k]=self.reWriteEmail(emails[k])
            k+=1
        cnt=0
        for _ in list(set(emails)):
            cnt+=1
        return cnt
    
    def reWriteEmail(self,email):
        rep=email.split('@')
        rep[0]=rep[0].replace('.','')
        k=0
        while k<len(rep[0]):
            if rep[0][k]!='+':
                k+=1
            else:
                break
        rep[0]=rep[0][:k]
        return rep[0]+'@'+rep[1]
